docker-images/grafana/cmd/grafana-wrapper/site.go

```diff
@@ -0,0 +1,220 @@
+		c.mux.RUnlock()
+
+		// ignore irrelevant changes
+		if isUnchanged {
+			c.log.Debug("config updated contained no relevant changes - ignoring")
+			return
+		}
+
+		// update configuration
+		c.updateGrafanaConfig(ctx, newSiteConfig.ObservabilityAlerts, newSum)
+	})
+}
+
+func (c *siteConfigSubscriber) resetSrcNotifiers(ctx context.Context) error {
+	alerts, err := c.grafana.GetAllAlertNotifications(ctx)
+	if err != nil {
+		return err
+	}
+	for _, alert := range alerts {
+		if strings.HasPrefix(alert.UID, "src-") {
+			if err := c.grafana.DeleteAlertNotificationUID(ctx, alert.UID); err != nil {
+				return fmt.Errorf("failed to delete alert %q: %w", alert.UID, err)
+			}
+		}
+	}
+	return nil
+}
+
+// updateGrafanaConfig updates grafanaAlertsSubscriber state and writes it to disk. It never returns an error,
+// instead all errors are reported as problems
+func (c *siteConfigSubscriber) updateGrafanaConfig(ctx context.Context, newAlerts []*schema.ObservabilityAlerts, newSum []byte) {
+	c.mux.Lock()
+	defer c.mux.Unlock()
+	c.log.Debug("updating grafana configuration")
+
+	var problems conf.Problems
+
+	// generate new notifiers configuration
+	created, err := generateNotifiersConfig(c.alerts, newAlerts)
+	if err != nil {
+		problems = append(problems, newObservabilityAlertsProblem(err))
+		return
+	}
+
+	// get the general alerts panels in the home dashboard
+	homeBoard, err := getOverviewDashboard()
+	if err != nil {
+		problems = append(problems, newObservabilityAlertsProblem(fmt.Errorf("failed to generate alerts overview dashboard: %w", err)))
+		return
+	}
+	var criticalPanel, warningPanel *sdk.Panel
+	for _, panel := range homeBoard.Panels {
+		var level string
+		switch panel.CommonPanel.Title {
+		case "Critical alerts by service":
+			level, criticalPanel = "critical", panel
+		case "Warning alerts by service":
+			level, warningPanel = "warning", panel
+		default:
+			continue
+		}
+		panel.Alert = newDefaultAlertsPanelAlert(level)
+	}
+	if criticalPanel == nil || warningPanel == nil {
+		problems = append(problems, newObservabilityAlertsProblem(errors.New("failed to find alerts panels")))
+		return
+	}
+
+	if err := c.resetSrcNotifiers(ctx); err != nil {
+		problems = append(problems, newObservabilityAlertsProblem(err))
+		// silently try to recreate alerts, in case any were deleted
+		c.log.Warn("failed to reset notifiers - attempting to recreate")
+		for _, alert := range created {
+			if _, err := c.grafana.CreateAlertNotification(ctx, alert); err != nil {
+				c.log.Warn(fmt.Sprintf("failed to recreate notifier %q", alert.UID), "error", err)
+			}
+		}
+		return
+	}
+	for _, alert := range created {
+		_, err = c.grafana.CreateAlertNotification(ctx, alert)
+		if err != nil {
+			c.log.Error(fmt.Sprintf("failed to create notifier %q", alert.UID), "error", err)
+			problems = append(problems, newObservabilityAlertsProblem(fmt.Errorf("failed to create alert %q: please refer to the Grafana logs for more details", alert.UID)))
```

Comment:
```suggestion
			problems = append(problems, newObservabilityAlertsProblem(fmt.Errorf("failed to create alert %q: please refer to the Grafana logs for more details", alert.UID)))
			problems = append(problems, newObservabilityAlertsProblem(fmt.Errorf("grafana error:", err)))
```

Admins will most likely send us screenshots of the error they get, so having this will be useful to identify the most common errors we see in the field.